### ISSUE TITLE
Bug report: FileNotFoundError due to special characters in pathnames at  /ultralytics/data/base.py

### ISSUE THREAD

**Issue by mpperez3** [ultralytics/ultralytics#13063]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

**Issue Description:**

When trying to load images from a directory that contains special characters (e.g., [, ]), the function get_img_files in ultralytics/data/base.py fails to find the images. The issue is due to not escaping special characters in the pathname using glob.escape(pathname).

**Method Definition /ultralytics/data/base.py:**

    def get_img_files(self, img_path):
        """Read image files."""
        try:
            f = []  # image files
            for p in img_path if isinstance(img_path, list) else [img_path]:
                p = Path(p)  # os-agnostic
                if p.is_dir():  # dir
                    f += glob.glob(str(p / "**" / "*.*"), recursive=True) <---- Empty list if p contains characters as '[' 
                    # F = list(p.rglob('*.*'))  # pathlib
                elif p.is_file():  # file
                    with open(p) as t:
                        t = t.read().strip().splitlines()
                        parent = str(p.parent) + os.sep
                        f += [x.replace("./", parent) if x.startswith("./") else x for x in t]  # local to global path
                        # F += [p.parent / x.lstrip(os.sep) for x in t]  # local to global path (pathlib)
                else:
                    raise FileNotFoundError(f"{self.prefix}{p} does not exist")
            im_files = sorted(x.replace("/", os.sep) for x in f if x.split(".")[-1].lower() in IMG_FORMATS)
            # self.img_files = sorted([x for x in f if x.suffix[1:].lower() in IMG_FORMATS])  # pathlib
            assert im_files, f"{self.prefix}No images found in {img_path}. {FORMATS_HELP_MSG}"
        except Exception as e:
            raise FileNotFoundError(f"{self.prefix}Error loading data from {img_path}\n{HELP_URL}") from e
        if self.fraction < 1:
            im_files = im_files[: round(len(im_files) * self.fraction)]  # retain a fraction of the dataset
        return im_files

**Problem Source:**
Upon reviewing the related Tuner class at [ /ultralytics/data/base.py](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/data/base.py) line 99, it appears that glob.glob(str(p / "**" / "*.*"), recursive=True) return empty if img_path contains characters such as '[' or ']' 

### Environment

ultralytics main branch
OS                  Linux
Environment         Conda
Python              3.10.11

### Minimal Reproducible Example

result = model.tune(data='data': PosixPath(".../'example[yolov9c][v1]'/train.yaml")", project=project_path/"tune",
           seed=1024)

### Additional

Recomendation, use **glob.escape(pathname)** such is recommended here: https://stackoverflow.com/a/63319226/1708774

**glob.glob(str(glob.escape(p) / "**" / "*.*"), recursive=True)**

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by mpperez3** [comment]
Occurs a similar bug at /ultralytics/nn/autobackend.py (line:82): 

    def __init__(
        self,
        weights="yolov8n.pt",
        device=torch.device("cpu"),
        dnn=False,
        data=None,
        fp16=False,
        batch=1,
        fuse=True,
        verbose=True,
    ):
        """
        Initialize the AutoBackend for inference.

        Args:
            weights (str): Path to the model weights file. Defaults to 'yolov8n.pt'.
            device (torch.device): Device to run the model on. Defaults to CPU.
            dnn (bool): Use OpenCV DNN module for ONNX inference. Defaults to False.
            data (str | Path | optional): Path to the additional data.yaml file containing class names. Optional.
            fp16 (bool): Enable half-precision inference. Supported only on specific backends. Defaults to False.
            batch (int): Batch-size to assume for inference.
            fuse (bool): Fuse Conv2D + BatchNorm layers for optimization. Defaults to True.
            verbose (bool): Enable verbose logging. Defaults to True.
        """
        super().__init__()
        w = str(weights[0] if isinstance(weights, list) else weights) <--- this transformation does not escape the final string path producing subsequent errors.

**Comment by Burhan-Q** [comment]
@mpperez3 I'm not seeing an issue with this example (which would use `Autobackend`).

```py
from ultralytics import YOLO

model = YOLO('[yolo](v8n).pt')
result = model.predict("ultralytics/assets/bus.jpg")

>>> image 1/1 /ultralytics/assets/bus.jpg: 640x480 4 persons, 1 bus, 1 stop sign, 95.8ms
>>> Speed: 3.6ms preprocess, 95.8ms inference, 540.0ms postprocess per image at shape (1, 3, 384, 640)

out = model.export(format="onnx", half=True)
print(out)
>>> '[yolo](v8n).onnx'

model = YOLO(out, task="detect")

result = model.predict("ultralytics/assets/bus.jpg")
>>> image 1/1 /ultralytics/assets/bus.jpg: 640x640 4 persons, 1 bus, 68.0ms
>>> Speed: 6.3ms preprocess, 84.1ms inference, 1.8ms postprocess per image at shape (1, 3, 640, 640)
```

Could you share an example of the command that causes the issue? Additionally can you test using one of the pretrained models?

**Comment by mpperez3** [comment]
@Burhan-Q thanks for your prompt attention!. I had left an example in the "Minimal Reproducible Example" section.

The issue is in the param "data" or "project" of the train method, not in the YOLO constructor. For example:
```python
results = model.train(data="/my/example/data[yolo](v8n)[alpha]/v1", epochs=100, imgsz=640, project="/my/example/results[yolo](v8n)[alpha]/v1")`
```
Also try the following to check if there is also a problem in the predict method:
```python
model.predict("/ultralytics/my/example/results[yolo](v8n)[alpha]/assets/bus.jpg")
```

**Comment by Burhan-Q** [comment]
@mpperez3 I see, thank you for pointing out your example. I was able to replicate this issue are you willing to open a PR to help resolve this?
